01_financial-well-being/housing-value.qmd

```diff
@@ -0,0 +1,904 @@
+puma_county_crosswalk <-
+  read_csv(here::here(
+    "geographic-crosswalks",
+    "data",
+    "crosswalk_puma_to_county.csv"
+  )) |>
+  rename(
+    state = statefip,
+  ) |>
+  filter(state != "72")
+
+
+households_county <-
+  acs1yr |>
+  right_join(puma_county_crosswalk,
+    by = c("state", "puma", "crosswalk_period")
+  ) |>
+  # if county isn't present in the ACS use county codes from PUMA-County crosswalk
+  mutate(
+    county = if_else(county.x != "000", county.x, county.y),
+    hhwt = if_else(county.x != "000", hhwt, hhwt * afact),
+    sum_products = if_else(county.x != "000", 1, sum_products)
+  ) |>
+  select(-c(county.x, county.y))
+```
+
+#### Place
+
+```{r}
+#| label: puma-to-place-crosswalk
+
+
+puma_place_crosswalk <-
+  read_csv(here::here(
+    "geographic-crosswalks",
+    "data",
+    "crosswalk_puma_to_place.csv"
+  )) |>
+  rename(state = statefip)
+
+households_place <-
+  acs1yr |>
+  right_join(puma_place_crosswalk,
+    by = c("state", "puma", "crosswalk_period")
+  ) |>
+  mutate(hhwt = hhwt * afact)
+```
+:::
+
+## Create Housing Value Metric {#sec-create-metric}
+
+### Housing Share
+
+We will now compute the share of households by race and ethnicity (i.e., total number of households for each race and ethnic group divided by the total number of households)
+
+::: panel-tabset
+### County
+
+```{r}
+#| label: share-of-households-by-race-ethnicity-county
+
+share_households_county <-
+  households_county |>
+  select(year, state, county, hhwt, race_ethnicity) |>
+  summarise(
+    household = sum(hhwt),
+    .by = c(year, state, county, race_ethnicity)
+  ) |>
+  mutate(
+    total = sum(household),
+    .by = c(year, state, county)
+  ) |>
+  mutate(share_households = household / total) |>
+  select(year, state, county, race_ethnicity, share_households)
+
+share_households_county_subgroup <-
+  households_county |>
+  select(year, state, county, hhwt, race_ethnicity, subgroup) |>
+  summarise(
+    household = sum(hhwt),
+    .by = c(year, state, county, race_ethnicity, subgroup)
+  ) |>
+  mutate(
+    total = sum(household),
+    .by = c(year, state, county, subgroup)
+  ) |>
+  mutate(share_households = household / total) |>
+  select(year, state, county, race_ethnicity, subgroup, share_households)
+```
+
+**Check:** Is `share_households` between 0 and 1?
+
+```{r}
+#| label: check-share-hh-county-check
+stopifnot(min(share_households_county$share_households) >= 0)
+stopifnot(max(share_households_county$share_households) <= 1)
+```
+
+We visualize the distribution of the share of households by race and ethnicity
+
+```{r}
+#| label: check-share-households-county
+
+share_households_county |>
+  drop_na() |>
+  ggplot(aes(x = share_households)) +
+  geom_histogram(color = "blue", fill = alpha("blue", 0.3), binwidth = 0.02) +
+  facet_wrap(. ~ race_ethnicity) +
+  labs(
+    title = "Share of Households by Race and Ethnicity",
+    subtitle = "County",
+    y = "Count",
+    x = "Share of Households"
+  )
+
```

Comment:
This visualization shows up in the "plots" window-- I'm not sure if you wanted the results embedded in the .qmd or not?

Reply:
The other visualizations also show up in the plot window-- some are harder to see this way than others. Maybe note that all plots will be shown there?

Reply:
I think this will depend on each user's RStudio settings.
<img width="559" alt="Screenshot 2024-03-05 at 1 49 47 PM" src="https://github.com/UI-Research/mobility-from-poverty/assets/56268747/089d146c-4c3c-461b-bbdd-05969c6e03ce">